### DISCUSSION TITLE
Migrating from Eagle to KiCad?

### DISCUSSION THREAD

**Discussion by Poofjunior** [harp-tech/protocol#84] [Open-ended]
Autodesk recently announced that they will ultimately shut down support for Eagle ([announcement page](https://www.autodesk.com/products/fusion-360/blog/future-of-autodesk-eagle-fusion-360-electronics/), [Hackaday article + conversation](https://hackaday.com/2023/06/09/they-used-to-be-a-big-shot-now-eagle-is-no-more/)). The choice looks motivated by having folks opt for their subscription-based integrated electromechanical package, Fusion 360.

I'm somewhat uncomfortable having design files derive from a software that has regularly changed its terms over the last decade. With the above announcement on the horizon, would it make sense to migrate to a package with a more open model like [KiCad](https://www.kicad.org/)? 

I know there's a bit of a learning curve involved in learning a new package, but KiCad is a stellar package, supporting fairly complex EDA features like (1) matched differential signal lengths, (2) push-and-shove routing, (3) STEP export, (4) DXF import, (5) bulk-component BOM editing, (6) integrated SPICE simulation from schematics, and (7) deriving schematics and PCB layouts from templates. The developers also maintain fairly rigorous [conventions for symbol and footprint libraries](https://klc.kicad.org/), some of which make pick-and-place assembly a fairly straightforward process. I made the switch from Eagle to KiCad back in 2015, and I haven't regretted it since.

What do folks think about switching to KiCAD for future Harp designs?

**Comment by artursilva0** [comment]
Currently Eagle is already part of the Fusion 360 CAD software subscription, so I'm not surprised that they will move into a more integrated solution with Fusion. It seems that the Fusion 360 electronics is more of a rebranding of Eagle, now cloud based (some of the menus are exactly the same).

This being said, I agree that over the recent years there were too many changes in their business model, from a subscription model to the bundle subscription with Fusion and now to the Fusion 360 electronics. KiCAD seems to be a solid option (and definitely more stable), but more important is free and an open source, which makes more sense in the context of the Harp development.

The challenge here is to move into a new solution, when you have several years of designs and libraries developed in Eagle. On top of that we have some electromechanically designs integrated with Fusion, which might be straightforward to adapt using a 3D generated model from KiCAD, but we will need to check on this.

This is a discussion that we have been having, and sooner or later the phase-out from Eagle would occur, but it would be a slow transition and it will depend a lot on the urgency of the next designs, since the ramp up on using a new software and mainly on recreating some of our custom libraries will surely take some additional time. 

Thanks for bringing this up!
